lib/rules/no-constant-condition.js

```diff
@@ -31,8 +31,14 @@ module.exports = {
                 type: "object",
                 properties: {
                     checkLoops: {
-                        type: "boolean",
-                        default: true
+                        oneOf: [
+                            {
+                                enum: ["all", "allExceptWhileTrue", "none"]
+                            },
+                            {
+                                type: "boolean"
+                            }
+                        ]
```

Comment:
```suggestion
                        enum: ["all", "allExceptWhileTrue", "none", true, false]
```

I think we can simplify the schema this way. We're already using this pattern in other rules, for example:

https://github.com/eslint/eslint/blob/113f51ec4e52d3082a74b9682239a6e28d1a70ee/lib/rules/no-mixed-spaces-and-tabs.js#L25-L29

Reply:
Done!